### PR TITLE
feat(codeblock/copybutton): swizzle CodeBlock/CopyButton to remove leading shell signs in codeblock

### PR DESCRIPTION
**Pull request by lloesche**
# Description

Swizzle CodeBlock/CopyButton to remove leading > and $ signs when copying to clipboard and completely remove lines starting with [`\u200b` (ZERO WIDTH SPACE)](https://unicode-table.com/en/200B/).

The updated code blocks in various files are mainly for testing the JS code. There's more docs to be updated once this is merged.

Added `patches/copybutton.patch/.sh` files to make it easy to reapply the swizzle on upgrades.

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

**Comment by lloesche** [comment]
@aquamatthias wdyt about using zero width space character as magic character to indicate that a line should be removed from clipboard? This would allow us to have any content (like output) inside the code block, but when copying the code block you only get the raw command, copy'paste ready.

File: patches/copybutton.patch [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+-    copy(code);
++    copy(
++      code
++        .split('\n')
++        .map((str) =>
++          str
++            .replace(/^>{3}\s*/, '')
++            .replace(/^[>$]\s*/, '')
++            .replace(/^\s+$/, '')
++            .replace(/^\u200b.*/, '')
```
Comment (aquamatthias) [inline patches/copybutton.patch:27]: why the `.*` ?

**Review by aquamatthias — APPROVED** [review]
One comment - otherwise lgtm

File: patches/copybutton.patch (hunk 1 above)
Comment (aquamatthias) [reply patches/copybutton.patch:27]: Ah saw your comment now.

Comment (aquamatthias) [reply patches/copybutton.patch:27]: I don't have a requirement for it yet - but maybe it is useful.

**Pull request merged**